lib/concurrent/synchronization/java_pure_object.rb

```diff
@@ -32,6 +33,10 @@ def ns_signal
           JRuby.reference0(self).notify
           self
         end
+
+        def ensure_ivar_visibility!
+          # relying on undocumented behavior of JRuby, ivar access is volatile
```

Comment:
I thought we worked out that added a new ivar that increased the size of the object was volatile, but not accessing an existing one?

But whether or not that's true - how about we simply make it documented behaviour by agreeing it with the JRuby team and getting it written down somewhere?